lib/expander/src/Events/CourseModuleCompleted.php

```diff
@@ -14,13 +9,11 @@ class CourseModuleCompleted extends  Event
     public function read(array $opts)
     {
 
-        global $DB;
-
-        $moduleid = $opts['contextinstanceid'];
 
-        $cm = $DB->get_record('course_modules',array('id' =>$moduleid));
-        $cmtype = $DB->get_record('modules',array('id' =>$cm->module));
+        $contextinstanceid = $opts['contextinstanceid'];
 
+        $cm = $this->repo->readModuleByContext($contextinstanceid);
```

Comment:
cm  means  course_module according to the moodle documentation. 

https://docs.moodle.org/dev/Module_visibility_and_display